data/pools/transactionPool.go

```diff
@@ -582,16 +593,23 @@ func (pool *TransactionPool) addToPendingBlockEvaluatorOnce(txgroup []transactio
 					stats.StopReason = telemetryspec.AssembleBlockFull
 				} else {
 					stats.StopReason = telemetryspec.AssembleBlockTimeout
+					// if the block is not full, it means that the above transaction made it to the block, so we want to add it here.
+					stats.ProcessingTime.AddTransaction(transactionGroupDuration)
```

Comment:
Do we care not to set ProcessingTime if ErrNoSpace?
Can't just set this once on line 590?

Reply:
no; because we want to have the "numbers" to match up.